public/index.js

```diff
@@ -1,62 +1,146 @@
-// on change store the value of folder form
-// send value to server as folder for url
 
-  const testFolder = $('#folder-form')
+let foldersArray
+let urlsArray
+const folderForm = $('#folder-form')
+const folderSelect = $('#folder-select')
+const dataSubmit = $('#submit-url')
+const folderSorter = $('#folder-sort')
 
-  let foldersArray = [{title: 'yellow', url:'llllll', 'popularity': 0, 'id': 1},
-  {title: 'yellow', url:'llllll', 'popularity': 2, folder: 'color', 'id': 3},
-  {title: 'blue', url:'llllll', 'popularity': 4, folder: 'color', 'id': 2},
-  {title: 'red', url:'llllll', 'popularity': 1, folder: 'color', 'id': 4}]
-
-  const urlList = (title) => {
-    const foundFolder = foldersArray.filter((folder) => {
-      return folder.folder === title
+$(document).ready(() =>{
+  fetch('/api/v1/folders', {
+    method: 'GET',
+  })
+    .then((data) => data.json())
+    .then((folders) => {
+      console.log(folders)
+      foldersArray = folders
+      folders.forEach((folder) => {
+        $('#folders').append(`<option value="${folder.folder_name}"/>`)
+      })
     })
-    console.log(foundFolder);
-    return urlLister(foundFolder)
-  }
+    .catch(error => console.log(error))
+})
 
-  const urlLister = (folderObject) => {
-    console.log(folderObject)
-    folderObject.forEach((object) => {
-      console.log(object.title)
-      $('.url-list').append(`<li>${object.url}, ${object.title}</li>`)
-      // $('.url-list').append(`<li> ITEM HERE </li>`)
+const getAllUrls = () =>{
+  fetch('/api/v1/urls', {
+    method: 'GET',
+  })
+    .then((data) => data.json())
+    .then((urls) => {
+      console.log(urls)
     })
-  }
+    .catch(error => console.log(error))
+}
 
-  testFolder
-  .on('change', (e) => {
-    e.preventDefault()
-    //on submit, grab value of folder select and use for server things
+const getUrlsByFolder = (folderId) =>{
+  return fetch(`/api/v1/folders/${folderId}/urls`, {
+  method: 'GET',
+  })
+  .then((data) => data.json())
+  .catch(error => console.log(error))
+}
 
-    $('.folder-title').replaceWith(`<div class = "folder-title"><h2>${e.target.value}</h2></div>`)
-    $('.folder-title').append(`<ul class = 'url-list'></ul>`)
-    $('.folder-content').addClass('active')
-    urlList(e.target.value)
+const addUrls = (folder, url, urlTitle) =>{
+  fetch('/api/v1/folders/', {
+    method: 'POST',
+    headers: {'Content-Type': 'application/json'},
+    body: JSON.stringify({
+      "folder_name": `${folder}`,
+      "original_url": `${url}`,
+      "title": `${urlTitle}`
+    })
   })
-  .on('submit', (e) => {
-    e.preventDefault();
+    .then((response) => console.log(response))
+    .catch(error => console.log(error))
+}
+
+folderForm
+.on('change', (e) => {
+  e.preventDefault()
+  let folderVal = folderSelect.val()
```

Comment:
I think these could be const
The content being assigned is dynamic, but not the assignment itself. Right? Or maybe I'll see something later on.